lib/elixir/test/elixir/kernel/warning_test.exs

```diff
@@ -28,6 +28,26 @@ defmodule Kernel.WarningTest do
     purge Sample
   end
 
+  test "unused variable in redefined function in different file" do
+    output = capture_err(fn ->
+      Code.eval_string """
+      defmodule Sample do
```

Comment:
Oh great. That is definitely better..
Pulling in Supervisor might make following the intention harder
What about this version?